### PR TITLE
Use cereal lib configured in WORKSPACE and remove tar.gz

### PR DESCRIPTION
**Pull request by nwangtw**

### CODE REVIEW

**Comment by nwangtw** [comment]
@kramasamy @srkukarni I saw you guys added the tar.gz file and changed the version from 1.2.2 to 1.2.1 in git log. Could you please take a look at this PR? And do I need to change 1.2.2 to 1.2.1?

Thanks in advance.

File: third_party/cereal/BUILD [hunk 1]
```diff
@@ -116,9 +116,7 @@ genrule(
 
 cc_library(
     name = "cereal-cxx",
-    srcs = [
-        "empty.cc",
```
Comment (nlu90) [inline third_party/cereal/BUILD]: IIRC, the empty file is needed for compiling correctly

Comment (nwangtw) [reply third_party/cereal/BUILD]: "bazel build --config=darwin heron/..." seems to be working ok for me locally. Doing another clean rebuild now.

Comment (nwangtw) [reply third_party/cereal/BUILD]: Builds ok.

INFO: Elapsed time: 1086.264s, Critical Path: 195.24s
INFO: 2515 processes: 2202 local, 313 worker.
INFO: Build completed successfully, 4264 total actions

**Comment by kramasamy** [comment]
@nlu90 - you don't need BUILD, *.tar.gz and empty.cc at all. All you need cereal.BUILD - nothing else.

**Comment by kramasamy** [comment]
Those files can be safely removed.

**Comment by kramasamy** [comment]
@nlu90 - the files BUILD, empty.cc can be removed.

**Pull request merged**